ci/docker_login.py

```diff
@@ -0,0 +1,137 @@
+
+    session = boto3.Session()
+    client = session.client(
+        service_name='secretsmanager',
+        region_name=secret_endpoint_region_name,
+        endpoint_url=secret_endpoint_url
+    )
+    try:
+        get_secret_value_response = client.get_secret_value(
+            SecretId=secret_name
+        )
+    except botocore.exceptions.ClientError as client_error:
+        if client_error.response['Error']['Code'] == 'ResourceNotFoundException':
+            logging.exception("The requested secret %s was not found", secret_name)
+        elif client_error.response['Error']['Code'] == 'InvalidRequestException':
+            logging.exception("The request was invalid due to:")
+        elif client_error.response['Error']['Code'] == 'InvalidParameterException':
+            logging.exception("The request had invalid params:")
+        raise
+    else:
+        secret = get_secret_value_response['SecretString']
+        secret_dict = json.loads(secret)
+        return secret_dict
+
+
+@retry(target_exception=subprocess.CalledProcessError, tries=DOCKERHUB_LOGIN_NUM_RETRIES,
+       delay_s=DOCKERHUB_RETRY_SECONDS)
+def login_dockerhub(secret_name: str, secret_endpoint_url: str, secret_endpoint_region_name: str):
+    """
+    Login to the Docker Hub account
+    :return: None
+    """
+    dockerhub_credentials = _get_dockerhub_credentials(secret_name, secret_endpoint_url, secret_endpoint_region_name)
+
+    logging.info('Logging in to DockerHub')
+    # We use password-stdin instead of --password to avoid leaking passwords in case of an error.
+    # This method will produce the following output:
+    # > WARNING! Your password will be stored unencrypted in /home/jenkins_slave/.docker/config.json.
+    # > Configure a credential helper to remove this warning. See
+    # > https://docs.docker.com/engine/reference/commandline/login/#credentials-store
+    # Since we consider the restricted slaves a secure environment, that's fine. Also, using this will require
+    # third party applications which would need a review first as well.
+    p = subprocess.run(['docker', 'login', '--username', dockerhub_credentials['username'], '--password-stdin'],
+                       stdout=subprocess.PIPE, input=str.encode(dockerhub_credentials['password']))
+    logging.info(p.stdout)
+    if p.returncode == 0:
+        logging.info('Successfully logged in to DockerHub')
+        return True
+
+    raise RuntimeError("Failed to login to DockerHub")
+
+
+def logout_dockerhub():
+    """
+    Log out of DockerHub to delete local credentials
+    :return: None
+    """
+    logging.info('Logging out of DockerHub')
+    subprocess.call(['docker', 'logout'])
+    logging.info('Successfully logged out of DockerHub')
+
+
+def main(sys_argv):
+    config_logging()
+
+    parser = argparse.ArgumentParser(
+        description="Safe docker login utility to avoid leaking passwords",
+        epilog=""
+    )
+    parser.add_argument("--secret-name",
```

Comment:
I won't put this through because we had a whole CoE on leaked docker passwords - this is why we ended up having to use the secrets manager =P I'm sure if I add it, Marco will flip... ._.